### PR TITLE
Renames palette story

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
Renames `Palette.story.tsx` (uppercase) so `.design` can detect it (import statements are case sensitive)

Resolves No. 6 https://docs.google.com/document/d/1zds271-DF_vmsF4-yy7jjV20jvpEtpTsSPIxf5w7xo4/edit

### CODE REVIEW

File: .changeset/tender-birds-flow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/palette': patch
```
Comment (brookescarlett) [inline .changeset/tender-birds-flow.md:2]: Not sure that we need a changeset but won't block on this!

**Review by brookescarlett — APPROVED** [review]

File: .changeset/tender-birds-flow.md (hunk 1 above)
Comment (TheSonOfThomp) [reply .changeset/tender-birds-flow.md:2]: I think we need a changeset to make sure it publishes

Comment (brookescarlett) [reply .changeset/tender-birds-flow.md:2]: Duh you're right.. you'll have to add a changeset to your typography PR then as well

Comment (TheSonOfThomp) [reply .changeset/tender-birds-flow.md:2]: Will do 👍

**Pull request merged**